File: src/main/java/project/rpg/commands/SkillCommand.java
package project.rpg.commands;

import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;
import org.jetbrains.annotations.NotNull;
import project.rpg.player.info.Skill;
import project.rpg.skill.magic.fire.MeteoStrike;
import project.rpg.skill.SkillDic;


public class SkillCommand implements CommandExecutor {
    @Override
    public boolean onCommand(@NotNull CommandSender sender, @NotNull Command command, @NotNull String label, @NotNull String[] args) {
        if (sender instanceof Player) {

            Player player = (Player) sender;

            if ("add".equals(args[0])) {

                if(SkillDic.isExist(args[1])) {
                    player.sendMessage(args[1] + " is added ");
                    Skill.addSkill(player,SkillDic.getSkill(player,args[1]));
                } else {
                    sender.sendMessage("오타난 커멘드");
                }

            } else {
                player.sendMessage(command.getUsage());
            }
            return true;
        }
        return false;
    }
}

File: src/main/java/project/rpg/commands/SkillCommand.kt
package project.rpg.commands

import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.command.TabCompleter
import org.bukkit.entity.Player
import project.rpg.player.info.Skill
import project.rpg.skill.SkillDic

class SkillCommand : CommandExecutor, TabCompleter {
    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<String>): Boolean {
        if (sender is Player) {
            if ("add" == args[0]) {
                if (args.size >= 2){
                    if (SkillDic.isExist(args[1])) {
                        Skill.addSkill(sender, SkillDic.makeSkill(sender, args[1]))
                        sender.sendMessage(args[1] + " is added ")
                    } else {
                        sender.sendMessage(command.usage)
                    }
            } else {
                sender.sendMessage(command.usage)
            }
        }
            return true
        }
        return false
    }

    override fun onTabComplete(
        sender: CommandSender,
        command: Command,
        label: String,
        args: Array<out String>?
    ): MutableList<String> {
        val recommendation = ArrayList<String>()

        if(args != null){
            when(args.size){
                1 -> recommendation.add("add")
                2 -> {
                    recommendation.addAll(SkillDic.getSkillList())
                }
            }
        }
        return recommendation
    }
}